04 - Poker.playground/Contents.swift

```diff
@@ -8,15 +8,174 @@
 /*
  EURE ANTWORT HIER
 */
+import Foundation
 
+enum Suit: Int, CustomStringConvertible  {
+    
+    case Karo, Herz, Pik, Kreuz
 
+    var description: String {
+        get {
+            
+            switch self{
+                
+            case .Karo:
+                return "\u{2666}"
+            case .Herz:
+                return "\u{2665}"
+            case .Pik:
+                return "\u{2660}"
+            case .Kreuz:
+                return "\u{2663}"
+                
+                
+            }
 
+            
+        }
+    }
+    
+    
+}
+
+enum Rank: Int {
+    
+    case two, three, four, five, six, seven, eight, nine, ten, bube, dame, könig, ass
+    
+    var description: String {
+        get {
+            
+            switch self{
+                
+            case .two:
+                return "2"
+            case .three:
+                return "3"
+            case .four:
+                return "4"
+            case .five:
+                return "5"
+            case .six:
+                return "6"
+            case .seven:
+                return "7"
+            case .eight:
+                return "8"
+            case .nine:
+                return "9"
+            case .ten:
+                return "10"
+            case .bube:
+                return "Bube"
+            case .dame:
+                return "Dame"
+            case .könig:
+                return "König"
+            case .ass:
+                return "ass"
+
+
+                
+                
+            }
+            
+            
+        }
+    }
+
+}
+
+enum Ranking: Int {
+    case highCard, flush, straightFlush, fourOfaKind, fullHouse, straight, threeOfaKind, twoPair, onePair
+    
+    var description: String {
+        switch self{
+        case .flush:
+            return "Flush"
+        case .highCard:
+            return "high Card"
+        default:
+            return "was anderes"
+        }
+    }
+    
+}
+
+struct Card: CustomStringConvertible, Equatable  {
+    
+    let suit: Suit
+    let rank: Rank
+    
+    var description: String {
+        get{
+            return "\(suit) \(rank)"
+        }
+    }
+    
+}
 
+struct PokerHand {
+    
+    var cards = [Card(suit: Suit.Karo, rank: Rank.ass), Card(suit: Suit.Karo, rank: Rank.ass), Card(suit: Suit.Karo, rank: Rank.ass), Card(suit: Suit.Karo, rank: Rank.ass), Card(suit: Suit.Karo, rank: Rank.ass)]
+    
```

Comment:
Schreib das Attribut lieber als Konstante und ohne Anfangswert. Dann in `init` mit einem neuen variablen Array arbeiten und es dann dem Attribut `self.cards` zuweisen.